File: lib/domain/public_plan/ports/entry_schedule_show_output_port.rb
# frozen_string_literal: true

module Domain
  module PublicPlan
    module Ports
      class EntryScheduleShowOutputPort
        def on_success(success_dto)
          raise NotImplementedError, "#{self.class.name} must implement #{__method__}"
        end
      end
    end
  end
end

File: crates/agrr-domain/src/public_plan/ports/entry_schedule_show_output_port.rs
use crate::public_plan::dtos::{EntryScheduleFailure, EntryScheduleShowOutput};

/// Ruby: `Domain::PublicPlan::Ports::EntryScheduleShowOutputPort`
pub trait EntryScheduleShowOutputPort {
    fn on_success(&mut self, success_dto: EntryScheduleShowOutput);
    fn on_failure(&mut self, failure_dto: EntryScheduleFailure);
}
